tools/clang/lib/Headers/hlsl/dx/linalg.h

```diff
@@ -0,0 +1,178 @@
+// Header for linear algebra APIs.
+
+#if ((__SHADER_TARGET_MAJOR > 6) ||                                            \
```

Comment:
Should we make this header error if it is included when targeting older shader models?
What about targeting SPIRV?

Reply:
The SPIRV path doesn't generate IR in DXC, it goes straight from the AST to SPIRV, which would likely cause the compiler either to crash or to emit a less-than helpful diagnostic.

Reply:
Maybe we could add something like this to the top of the file?

```hlsl
#if __spirv__
#error "Cooperative matrix not (yet) supported for SPIRV"
#endif
```

Reply:
s/matrix/vector